R/interval_collapse.R

```diff
@@ -137,7 +137,7 @@ collapse_common_intervals <- function(dt,
   # check for overlapping intervals
   if (overlapping_dt_severity != "skip") {
     overlapping_dt <- dt[
-      , identify_overlapping_intervals(unique(.SD)),
+      , identify_overlapping_intervals(unique(.SD), identify_all_possible = overlapping_dt_severity != "none"),
```

Comment:
The error message on 151 is only triggered if `overlapping_dt_severity != "none"`. We want the error message to show all possible overlapping intervals.

But if the error is skipped then line 155 tries to drop the least granular overlapping interval.